controllers/integration/integration_adapter.go

```diff
@@ -0,0 +1,310 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	hasv1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/controllers/results"
+	"github.com/redhat-appstudio/integration-service/tekton"
+	releasev1alpha1 "github.com/redhat-appstudio/release-service/api/v1alpha1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+// Adapter holds the objects needed to reconcile a Release.
+type Adapter struct {
+	pipelineRun *tektonv1beta1.PipelineRun
+	component   *hasv1alpha1.Component
+	application *hasv1alpha1.Application
+	logger      logr.Logger
+	client      client.Client
+	context     context.Context
+}
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(pipelineRun *tektonv1beta1.PipelineRun, component *hasv1alpha1.Component, application *hasv1alpha1.Application, logger logr.Logger, client client.Client,
+	context context.Context) *Adapter {
+	return &Adapter{
+		pipelineRun: pipelineRun,
+		component:   component,
+		application: application,
+		logger:      logger,
+		client:      client,
+		context:     context,
+	}
+}
+
+// EnsureApplicationSnapshotExists is an operation that will ensure that an integration ApplicationSnapshot associated
+// to the PipelineRun being processed exists. Otherwise, it will create a new integration ApplicationSnapshot.
+func (a *Adapter) EnsureApplicationSnapshotExists() (results.OperationResult, error) {
+	existingApplicationSnapshot, err := a.findMatchingApplicationSnapshot()
+	if err != nil {
+		return results.RequeueWithError(err)
+	}
+
+	if existingApplicationSnapshot == nil {
```

Comment:
I'd change it to an early exit instead of using an else clause. Negate it, log and continue processing.